black.py

```diff
@@ -2357,6 +2376,44 @@ def is_python36(node: Node) -> bool:
     return False
 
 
+def get_future_imports(node: Node) -> Set[str]:
+    """Returns a set of __future__ imports in the file."""
+    imports = set()
+    for child in node.children:
+        if child.type != syms.simple_stmt:
+            break
+        first_child = child.children[0]
+        if isinstance(first_child, Leaf):
+            # Continue looking if we see a docstring; otherwise stop.
+            if (
+                len(child.children) == 2
+                and first_child.type == token.STRING
+                and isinstance(child.children[1], Leaf)
+                and child.children[1].type == token.NEWLINE
+            ):
+                continue
+            else:
+                break
+        elif isinstance(first_child, Node) and first_child.type == syms.import_from:
+            module_name = first_child.children[1]
+            assert isinstance(module_name, Leaf)
```

Comment:
That might not be true. For example:
```
from a.b.c import X
```
a.b.c here is a Node of type `syms.dotted_name`.

Fold this into the `if` check below.

Reply:
Good catch, I added a test case for this.